Test/test21/issue-735.bpl

```diff
@@ -0,0 +1,18 @@
+// RUN: %parallel-boogie /proverOpt:O:smt.mbqi=true /timeLimit:5 /typeEncoding:p "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+// RUN: %parallel-boogie /proverOpt:O:smt.mbqi=true /typeEncoding:a "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+// RUN: %parallel-boogie /proverOpt:O:smt.mbqi=true /typeEncoding:m "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+type ref;
+type Field _;
+type Heap = [ref]<alpha>[Field alpha]alpha;
+function $IsGoodHeap(Heap) : bool;
+function $IsHeapAnchor(Heap) : bool;
+var $Heap: Heap where $IsGoodHeap($Heap) && $IsHeapAnchor($Heap);
```

Comment:
Is this variable necessary for the reproduction? Also I'm guessing we only need one of the two functions.

Reply:
Yeah, I suspect it could be simplified further. But it's relatively small and understandable now.